Engine/HostedAnalyzer.cs

```diff
@@ -0,0 +1,822 @@
+        /// <returns>The formatted script.</returns>
+        public string Format(string scriptDefinition)
+        {
+            Settings settings = CreateSettings("CodeFormatting");
+            string formattedScript;
+            lock(hostedAnalyzerLock)
+            {
+                try {
+                    formattedScript = Formatter.Format(scriptDefinition, settings, null, ps.Runspace, writer);
+                }
+                finally {
+                    analyzer.CleanUp();
+                    // Reset is required because formatting leaves a number of settings behind which
+                    // should be cleared.
+                    Reset();
+                }
+            }
+            return formattedScript;
+        }
+
+        /// <summary>
+        /// Format a script according to the formatting rules
+        ///     PSPlaceCloseBrace
+        ///     PSPlaceOpenBrace
+        ///     PSUseConsistentWhitespace
+        ///     PSUseConsistentIndentation
+        ///     PSAlignAssignmentStatement
+        ///     PSUseCorrectCasing
+        /// </summary>
+        /// <param name="scriptDefinition">The script to format.</param>
+        /// <returns>The formatted script.</returns>
+        public Task<string> FormatAsync(string scriptDefinition) =>
+            Task.Run(() => Format(scriptDefinition));
+
+        /// <summary>
+        /// Format a script according to the formatting rules
+        ///     PSPlaceCloseBrace
+        ///     PSPlaceOpenBrace
+        ///     PSUseConsistentWhitespace
+        ///     PSUseConsistentIndentation
+        ///     PSAlignAssignmentStatement
+        ///     PSUseCorrectCasing
+        /// </summary>
+        /// <param name="scriptDefinition">The script to format.</param>
+        /// <param name="range">The range of the script to format</param>
+        /// <returns>The formatted script.</returns>
+        public string Format(string scriptDefinition, Range range)
+        {
+            Settings settings = CreateSettings("CodeFormatting");
+            string formattedScript;
+            lock(hostedAnalyzerLock)
+            {
+                try {
+                    formattedScript = Formatter.Format(scriptDefinition, settings, range, ps.Runspace, writer);
+                }
+                finally {
+                    analyzer.CleanUp();
+                    // Reset is required because formatting leaves a number of settings behind which
+                    // should be cleared.
+                    Reset();
+                }
+            }
+            return formattedScript;
+        }
+
+        /// <summary>
+        /// Format a script according to the formatting rules
+        ///     PSPlaceCloseBrace
+        ///     PSPlaceOpenBrace
+        ///     PSUseConsistentWhitespace
+        ///     PSUseConsistentIndentation
+        ///     PSAlignAssignmentStatement
+        ///     PSUseCorrectCasing
```

Comment:
the exact list of those rules could change, I'd take them out of the comment therefore.